src/Database/Redis/Cluster.hs

```diff
@@ -266,10 +267,10 @@ evaluatePipeline shardMapVar refreshShardmapAction conn requests = do
         resps <- concat <$>
           mapM (\(resp, (cc, r)) -> case resp of
               Right v -> return v
-              Left (e :: SomeException) ->
-                case displayException e of
-                  "TimeoutException" -> throwIO TimeoutException
-                  _ -> executeRequests (getRandomConnection cc conn) r
+              Left (err :: SomeException) ->
+                if isPrefixOf "TimeoutException" (displayException err)
```

Comment:
Do we need to match string in case of exception ? Can't we have a type of exception and pattern match it ?